### PR TITLE
Vision2.0

### PR DESCRIPTION
**Pull request by AustinFortunato**
Super cool!!! I hope it works

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 1]
```diff
@@ -154,13 +161,36 @@ public CommandBase slowTurn(boolean isTurningRight) {
...
             } else {
                 driveTrain.arcadeDrive(0, -0.4);
-
             }
         }, this);
         cmd.setName("Turning");
         return cmd;
     }
 
+        public CommandBase visionAlignDegreeser() {
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Drive.java:170]: I'm sorry, what is a "degreeser"?

File: src/vision/camera_debug.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import visionfunctions as vision
+
+vision.camera_init()
```
Comment (msoucy) [inline src/vision/camera_debug.py:3]: Debug programs?

File: src/vision/constants.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+# Filter Variables
+LOWER_COLOR = np.array([39, 92, 18])
+UPPER_COLOR = np.array([89, 255, 255])
+
+ANGLE_HIGH_THRESHOLD = 80
+ANGLE_LOW_THRESHOLD = 30
+
+# Globals
+(
```
Comment (msoucy) [inline src/vision/constants.py:37]: Why are you using globals in the code? Do you want to make puppies cry?

File: src/vision/robot.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import pyrealsense2 as rs2
```
Comment (msoucy) [inline src/vision/robot.py:1]: What does this file do and why does it import so much that we don't use?

File: src/vision/stream_debug.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import cv2
```
Comment (msoucy) [inline src/vision/stream_debug.py:1]: More debug scripts?

File: src/vision/visionfunctions.py [hunk 6]
```diff
@@ -27,29 +71,117 @@ def swapStream(isShooting, s):
...
+        line_img = np.zeros((HEIGHT, WIDTH, 3), np.uint8)
     # Find lines in selected image
-    lines = cv2.HoughLinesP(MED_EDGES, 1, radians(0.5), 25, maxLineGap=25)
+    lines = cv2.HoughLinesP(med_edges, 1, radians(0.5), 25, maxLineGap=25)
+
     return lines
+
+
+def line_filter(lines):
+    if lines is not None:
```
Comment (msoucy) [inline src/vision/visionfunctions.py:104]: Why would `lines` be `None` here?

File: src/vision/visionfunctions.py [hunk 7]
```diff
@@ -27,29 +71,117 @@ def swapStream(isShooting, s):
...
+                        new_line,
+                        x1,
+                        y1,
+                        x2,
+                        y2,
+                        x_total,
+                        y_total,
+                    )
+            elif ANGLE_HIGH_THRESHOLD > abs(new_slope) > ANGLE_LOW_THRESHOLD:
+                x_total, y_total = newLine(
```
Comment (msoucy) [inline src/vision/visionfunctions.py:130]: See if you can combine this with the above

**Review by msoucy — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Drive.java (hunk 1 above)
Comment (AustinFortunato) [reply src/main/java/frc/robot/subsystems/Drive.java:170]: Changed in the new PR

File: src/vision/camera_debug.py (hunk 2 above)
Comment (AustinFortunato) [reply src/vision/camera_debug.py:3]: All of thses py files are quinn things- they kind of just found their way into the pull request

**Comment by AustinFortunato** [comment]
Moving to the EvenMoreVision branch- le pull request is closed

**Pull request closed without merging**